compiler  incorrectly suggest converting `&x` ->`&mut x` instead of `&(mut x)` when matching on a `&T(T : Copy)`

### Code

```Rust
fn myfun(s : &[u32]) {
    for &x in s {
        while x < 1000 {
            x *= 2;
        }
        println!("{x}")
    }
}
```

### Current output

```Shell
error[E0384]: cannot assign twice to immutable variable `x`
 --> src/lib.rs:6:13
  |
4 |     for &x in s {
  |          - first assignment to `x`
5 |         while x < 1000 {
6 |             x *= 2;
  |             ^^^^^^ cannot assign twice to immutable variable
  |
help: consider making this binding mutable
  |
4 |     for &mut x in s {
  |          +++
help: to modify the original value, take a borrow instead
  |
4 |     for &ref mut x in s {
  |          +++++++

For more information about this error, try `rustc --explain E0384`.
```

### Desired output

```Shell
error[E0384]: cannot assign twice to immutable variable `x`
 --> src/lib.rs:6:13
  |
4 |     for &x in s {
  |          - first assignment to `x`
5 |         while x < 1000 {
6 |             x *= 2;
  |             ^^^^^^ cannot assign twice to immutable variable
  |
help: consider making this binding mutable
  |
4 |     for &(mut x) in s {
  |          ++++  +
help: you cannot modify the original value as it is behind a shared reference. use a different iterating method if you want to do so

For more information about this error, try `rustc --explain E0384`.
```

### Rationale and extra context

not sure if the second `help` i put is that useful, could be reworded or removed. 

### Other cases

```Rust

```

### Rust Version

```Shell
rustc 1.91.0 (f8297e351 2025-10-28)
binary: rustc
commit-hash: f8297e351a40c1439a467bbbb6879088047f50b3
commit-date: 2025-10-28
host: x86_64-pc-windows-msvc
release: 1.91.0
LLVM version: 21.1.2
```

### Anything else?

_No response_

<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs
+++ b/compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs
@@ -3940,14 +3940,16 @@ impl<'infcx, 'tcx> MirBorrowckCtxt<'_, 'infcx, 'tcx> {
         if let Some(decl) = local_decl
             && decl.can_be_made_mutable()
         {
-            let message = if matches!(
-                decl.local_info(),
-                LocalInfo::User(BindingForm::Var(VarBindingForm {
-                    opt_match_place: Some((_, match_span)),
-                    ..
-                })) if matches!(match_span.desugaring_kind(), Some(DesugaringKind::ForLoop))
-            ) && let Ok(binding_name) =
-                self.infcx.tcx.sess.source_map().span_to_snippet(decl.source_info.span)
+            let is_for_loop = matches!(
+                            decl.local_info(),
+                            LocalInfo::User(BindingForm::Var(VarBindingForm {
+                                opt_match_place: Some((_, match_span)),
+                                ..
+                            })) if matches!(match_span.desugaring_kind(), Some(DesugaringKind::ForLoop))
+            );
+            let message = if is_for_loop
+                && let Ok(binding_name) =
+                    self.infcx.tcx.sess.source_map().span_to_snippet(decl.source_info.span)
             {
                 format!("(mut {}) ", binding_name)
             } else {
@@ -3961,6 +3963,7 @@ impl<'infcx, 'tcx> MirBorrowckCtxt<'_, 'infcx, 'tcx> {
             );
 
             if !from_arg
+                && !is_for_loop
                 && matches!(
                     decl.local_info(),
                     LocalInfo::User(BindingForm::Var(VarBindingForm {

--- a/compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs
+++ b/compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs
@@ -3940,12 +3940,26 @@ impl<'infcx, 'tcx> MirBorrowckCtxt<'_, 'infcx, 'tcx> {
         if let Some(decl) = local_decl
             && decl.can_be_made_mutable()
         {
+            let message = if matches!(
+                decl.local_info(),
+                LocalInfo::User(BindingForm::Var(VarBindingForm {
+                    opt_match_place: Some((_, match_span)),
+                    ..
+                })) if matches!(match_span.desugaring_kind(), Some(DesugaringKind::ForLoop))
+            ) && let Ok(binding_name) =
+                self.infcx.tcx.sess.source_map().span_to_snippet(decl.source_info.span)
+            {
+                format!("(mut {}) ", binding_name)
+            } else {
+                "mut ".to_string()
+            };
             err.span_suggestion_verbose(
                 decl.source_info.span.shrink_to_lo(),
                 "consider making this binding mutable",
-                "mut ".to_string(),
+                message,
                 Applicability::MachineApplicable,
             );
+
             if !from_arg
                 && matches!(
                     decl.local_info(),

--- a/tests/ui/borrowck/borrowck-for-loop-deref-pattern-assignment.rs
+++ b/tests/ui/borrowck/borrowck-for-loop-deref-pattern-assignment.rs
@@ -1,0 +1,10 @@
+//! regression test for <https://github.com/rust-lang/rust/issues/148467>
+//! Ensure the diagnostic suggests `for &(mut x) ...` (parenthesized) instead of `&mut x`.
+
+fn main() {
+    let nums: &[u32] = &[1, 2, 3];
+    for &num in nums {
+        num *= 2; //~ ERROR cannot assign twice to immutable variable `num`
+        println!("{num}");
+    }
+}

--- a/tests/ui/borrowck/borrowck-for-loop-deref-pattern-assignment.stderr
+++ b/tests/ui/borrowck/borrowck-for-loop-deref-pattern-assignment.stderr
@@ -1,0 +1,20 @@
+error[E0384]: cannot assign twice to immutable variable `num`
+  --> $DIR/borrowck-for-loop-deref-pattern-assignment.rs:7:9
+   |
+LL |     for &num in nums {
+   |          --- first assignment to `num`
+LL |         num *= 2;
+   |         ^^^^^^^^ cannot assign twice to immutable variable
+   |
+help: consider making this binding mutable
+   |
+LL |     for &(mut num) num in nums {
+   |          +++++++++
+help: to modify the original value, take a borrow instead
+   |
+LL |     for &ref mut num in nums {
+   |          +++++++
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0384`.

--- a/tests/ui/borrowck/borrowck-for-loop-deref-pattern-assignment.stderr
+++ b/tests/ui/borrowck/borrowck-for-loop-deref-pattern-assignment.stderr
@@ -10,10 +10,6 @@ help: consider making this binding mutable
    |
 LL |     for &(mut num) num in nums {
    |          +++++++++
-help: to modify the original value, take a borrow instead
-   |
-LL |     for &ref mut num in nums {
-   |          +++++++
 
 error: aborting due to 1 previous error
 

